tests/rustworkx_tests/generators/test_ladder.py

```diff
@@ -0,0 +1,26 @@
+# Licensed under the Apache License, 
+
+import unittest
+
+import rustworkx
+
+
+class TestLadderGraph(unittest.TestCase):
+    def test_ladder_graph(self):
+        graph = rustworkx.generators.ladder_graph(20)
+        self.assertEqual(len(graph), 40)
+        self.assertEqual(len(graph.edges()), 58)
+    
+    def test_ladder_graph_weights(self):
+        graph = rustworkx.generators.ladder_graph(weights=list(range(40)))
+        self.assertEqual(len(graph), 40)
+        self.assertEqual([x for x in range(40)], graph.nodes())
+        self.assertEqual(len(graph.edges()), 58)
```

Comment:
For these 2 tests it'd be good to add some assertion on the structure of the edges too. For example, if I built a circuit like:

```python
from rustworkx import rx
graph = rx.PyGraph()
graph.add_nodes_from(range(40))
graph.add_edges_from_no_data([(0, 1)*50)
```

We should try to make the tests so they assert the structure of the created graph is correct and if it were to change by mistake on a future change to the code the test will fail if it's no longer a ladder graph.